rust/kernel/str.rs

```diff
@@ -11,7 +11,69 @@ use crate::c_types;
 /// Byte string without UTF-8 validity guarantee.
 ///
 /// `BStr` is simply an alias to `[u8]`, but has a more evident semantical meaning.
-pub type BStr = [u8];
+#[repr(transparent)]
+pub struct BStr([u8]);
+
+impl BStr {
+    /// Returns the length of this string.
+    #[inline]
+    pub const fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Returns `true` if the string is empty.
+    #[inline]
+    pub const fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Creates a [`BStr`] from a `[u8]`.
+    #[inline]
+    pub const fn from_bytes(bytes: &[u8]) -> &BStr {
+        // SAFETY: BStr is transparent to [u8]
+        unsafe { core::mem::transmute(bytes) }
```

Comment:
`&*(bytes as *const [u8] as *const BStr)` looks better because it is more explicit.

Reply:
Thank you, done!